modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx

```diff
@@ -0,0 +1,265 @@
+				const getDeleteObjectDefinition = async () => {
+					const url = createResourceURL(baseResourceURL, {
+						objectDefinitionId: itemData.id,
+						p_p_resource_id:
+							'/object_definitions/get_object_definition_delete_info',
+					}).href;
+
+					const {
+						hasObjectRelationship,
+						objectEntriesCount,
+					} = await API.fetchJSON<{
+						hasObjectRelationship: boolean;
+						objectEntriesCount: number;
+					}>(url);
+
+					if (itemData.status.code !== 0) {
+						await deleteObjectDefinition(
+							itemData.id,
+							itemData.name
+						);
+						setTimeout(() => window.location.reload(), 1000);
+
+						return;
+					}
+
+					setDeletedObjectDefinition({
+						...itemData,
+						hasObjectRelationship,
+						objectEntriesCount,
+					});
+
+					setShowModal(
+						(previousState: ViewObjectDefinitionsModals) => ({
+							...previousState,
+							deleteObjectDefinition: true,
+						})
+					);
+				};
+
+				getDeleteObjectDefinition();
+			}
+		},
+		portletId:
+			'com_liferay_object_web_internal_object_definitions_portlet_ObjectDefinitionsPortlet',
+		sorting,
+		style: 'fluid' as 'fluid',
+		views: [
+			{
+				contentRenderer: 'table',
+				label: 'Table',
+				name: 'table',
+				schema: {
+					fields: [
+						{
+							contentRenderer:
+								'objectDefinitionLabelDataRenderer',
+							expand: false,
+							fieldName: 'label',
+							label: Liferay.Language.get('label'),
+							localizeLabel: true,
+							sortable: true,
+						},
+						{
+							expand: false,
+							fieldName: 'scope',
+							label: Liferay.Language.get('scope'),
+							localizeLabel: true,
+							sortable: false,
+						},
+						{
+							contentRenderer:
+								'objectDefinitionSystemDataRenderer',
+							expand: false,
+							fieldName: 'system',
+							label: Liferay.Language.get('system'),
+							localizeLabel: true,
+							sortable: false,
+						},
+						{
+							contentRenderer:
+								'objectDefinitionModifiedDateDataRenderer',
+							expand: false,
+							fieldName: 'dateModified',
+							label: Liferay.Language.get('modified-date'),
+							localizeLabel: true,
+							sortable: true,
+						},
+						{
+							contentRenderer:
+								'objectDefinitionStatusDataRenderer',
+							expand: false,
+							fieldName: 'status',
+							label: Liferay.Language.get('status'),
+							localizeLabel: true,
+							sortable: false,
+						},
+					],
+				},
+				thumbnail: 'table',
+			},
+		],
+	};
+
+	useEffect(() => {
+		Liferay.on('addObjectDefinition', () =>
+			setShowModal((previousState: ViewObjectDefinitionsModals) => ({
+				...previousState,
+				addObjectDefinition: true,
+			}))
+		);
+
+		return () => {
+			Liferay.detach('addObjectDefinition');
+		};
+	}, []);
+
+	return (
+		<>
+			<FrontendDataSet {...dataSetProps} />
+
+			{showModal.addObjectDefinition && (
+				<ModalAddObjectDefinition
+					apiURL={apiURL as string}
+					onVisibilityChange={() => {
```

Comment:
@rebsilva I think we can change the name of this prop to something like "handleClose" or "onClose" because it's being used to close the modal.